lib/types.js

```diff
@@ -38,6 +38,43 @@ module.exports.generateTypeConversion = function (name, idlType, argAttrs, custo
     }
     ${name} = V;
   }`;
+  } else if (idlType.generic === "record") {
+    if (!handleNullable || !idlType.nullable) {
+      str += `
+  if (${name} === null && ${name} === undefined) {
+    ${name} = new Map;
+  } else {`;
+    }
+
+    str += `
+  if (typeof ${name} !== "object") {
+    throw new TypeError("The value provided is not an object");
+  } else {
+    const result = new Map;
+    const keys = Object.keys(${name});
```

Comment:
Per spec this should be Object.getOwnPropertyKeys, and thne you should use getOwnPropertyDescriptor, and check enumerability, and so on. This is important as it impacts how many side effects occur before an exception happens in exceptional cases.